NDVItimeseries.R

```diff
@@ -0,0 +1,527 @@
+    legend.position="bottom"
+  ) 
+
+# Don't show scientific notation
+options(scipen = 999)
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Load Libraries and Options -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+library(tidyverse)
+library(haven) # for read_dta
+library(tidylog) # for output on operations/changes with select, drop, merges, etc.
+library(rio)
+library(janitor)
+library(sf) # for st_read
+
+library(openxlsx)
+library(readxl)
+library(moderndive) 
+library(readr)
+library(tidyr)
+library(lubridate) # for dates
+
+library(reshape2)
+library(ggplot2) # included in tidyverse
+library(ggthemes)
+library(stringr)
+library(dplyr)
+library(openxlsx)
+library(readr)
+library(sf)
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Load Data -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+#admin 2
+
+NDVI1admin2 <-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/NDVIMean_ts_niger (1981-2000).csv")
+
+NDVI2admin2<-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/NDVIMean_ts_niger (2001-2022).csv")
+
+#admin 3
+
+NDVI1admin3 <-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/l3NDVIMean_ts_niger (1981-2000).csv")
+
+NDVI2admin3 <- read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/l3NDVIMean_ts_niger (2001-2022).csv")
+
+#merge datasets
+NDVIadmin2 <- dplyr::left_join(NDVI1admin2, NDVI2admin2)
+NDVIadmin3 <- dplyr::left_join(NDVI1admin3, NDVI2admin3)
+
+# geospatial
+nigerMapAdmin2 <- 
+  st_read("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/Level 2 (1)/wb_niger_admin2_shapefile/niger_admin2.shp") 
+
+
+nigerMapAdmin3 <- 
+  st_read("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/Level 3 (1)/NER_adm03_feb2018.shp")
+
+
+# Don't show scientific notation
+options(scipen = 999)
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Clean titles and data -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("_NDVI", "", colnames(NDVIadmin2)[col])
+}
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("X", "", colnames(NDVIadmin2)[col])
+}
+
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("_NDVI", "", colnames(NDVIadmin3)[col])
+}
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("X", "", colnames(NDVIadmin3)[col])
+}
+head(NDVIadmin3)
+
+
+# Convert data from Wide to Long ---- 
+ndviData2 <- 
+  NDVIadmin2 %>%
+  select(admin1Name, admin1Pcod, 
+         admin2Name, admin2Pcod, "19810701": "20191231") 
```

Comment:
For greater reproducibility/ability to extend the analysis in the future, would identify the columns that have integers/dates and only select those, rather than particular dates. That could involve a function like ```select_if(df, ~is.numeric(.x)``` You may also need/want to add in a line earlier to capture the other vars that you want to include, i.e.  myvars <- c(admin1Name....). [This is bonus so if it takes more than 30 minutes to do, don't worry about and we'll address it later. But do keep an eye out for places where you 'hardcode' in elements of the data and consider alternatives that are more general/capture the characteristics of what you're looking for].